src/FLSonar.cc

```diff
@@ -602,6 +611,24 @@ void FLSonar::GenerateTransferTable(std::vector<int> &_transfer)
   }
 }
 
+//////////////////////////////////////////////////
+sonar_msgs::SonarStamped FLSonar::SonarRosMsg(gazebo::physics::WorldPtr _world)
```

Comment:
I believe that the `gazebo::physics::WorldPtr` getter methods are constants, thus you could pass the pointer `world` as a `const` reference.

Reply:
I agree, and I have done this modification